src/main/java/com/exasol/bucketfs/ReadEnabledBucket.java

```diff
@@ -118,45 +123,82 @@ protected String extendPathInBucketDownToFilename(final Path localPath, final St
-            throw new BucketAccessException("Unable to download file \"" + localPath + "\" from ", uri, exception);
-        }
+        requestFileOnBucket(uri, localPath);
         LOGGER.fine(() -> "Successfully downloaded file to \"" + localPath + "\"");
     }
 
-    // [impl->dsn~downloading-a-file-from-a-bucket-as-string~1]
-    @Override
-    public String downloadFileAsString(final String pathInBucket) throws InterruptedException, BucketAccessException {
-        final URI uri = createPublicReadURI(pathInBucket);
-        LOGGER.fine(() -> "Downloading  file from bucket \"" + this + "\" at \"" + uri + "\"");
+    private void requestFileOnBucket(final URI uri, final Path localPath) throws BucketAccessException {
         try {
-            return httpGet(uri);
+            final var request = createRequest(uri);
+            final var response = this.client.send(request, BodyHandlers.ofFile(localPath));
+            evaluateRequestStatus(uri, DOWNLOAD, response.statusCode());
         } catch (final IOException exception) {
-            throw new BucketAccessException("Unable to download file from BucketFS as string.", uri, exception);
+            throw createDownloadIoException(uri, DOWNLOAD, exception);
+        } catch (final InterruptedException exception) {
+            Thread.currentThread().interrupt();
+            throw createDownloadInterruptedException(uri, DOWNLOAD);
         }
     }
 
-    private String httpGet(final URI uri) throws IOException, InterruptedException {
-        final HttpRequest request = HttpRequest.newBuilder(uri) //
+    private HttpRequest createRequest(final URI uri) {
+        return HttpRequest.newBuilder(uri) //
                 .GET() //
                 .header("Authorization", encodeBasicAuthForReading()) //
                 .build();
-        final HttpResponse<String> response = this.client.send(request, BodyHandlers.ofString());
-        checkHttpStatusCode(response.statusCode());
+    }
+
+    private BucketAccessException createDownloadIoException(final URI uri, final BucketOperation operation,
+            final IOException exception) {
+        return new BucketAccessException(messageBuilder("E-BFSJ-5")
+                .message("I/O error trying to {{operation|uq}} {{URI}}", operation, uri).toString(), exception);
+    }
+
+    private BucketAccessException createDownloadInterruptedException(final URI uri, final BucketOperation operation) {
+        return new BucketAccessException(messageBuilder("E-BFSJ-4")
+                .message("Interrupted trying to {{operation|uq}} {{URI}}.", operation, uri).toString());
+    }
+
+    // [impl->dsn~downloading-a-file-from-a-bucket-as-string~1]
+    @Override
+    public String downloadFileAsString(final String pathInBucket) throws BucketAccessException {
+        final var uri = createPublicReadURI(pathInBucket);
+        LOGGER.fine(() -> "Downloading  file from bucket \"" + this + "\" at \"" + uri + "\"");
```

Comment:
I think it's better to use single quotes in log messages too to be consistent.

Reply:
Done.